File: BackEnd/src/main/java/com/covoiturage/covoiturage/models/Trajet.java
package com.covoiturage.covoiturage.models;

import com.fasterxml.jackson.annotation.JsonFormat;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.mongodb.core.mapping.Document;
import java.util.Date;



@Document(collection = "trajet")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Trajet {
    @Id
    private String id;
    private String conducteurId;
    private String villeDepart;
    private String villeArrivee;
    @JsonFormat(pattern = "dd/MM/yyyy")
    private Date date;
    private String heure;
    private int placesDisponibles;
    private double prix;
    private String voiture;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'")
    private Date timestamp;


    // Constructor without ID (useful for new Trajet creation)
    public Trajet(String conducteurId, String villeDepart, String villeArrivee, Date date, String heure, int placesDisponibles, double prix, String voiture, Date timestamp) {
        this.conducteurId = conducteurId;
        this.villeDepart = villeDepart;
        this.villeArrivee = villeArrivee;
        this.date = date;
        this.heure = heure;
        this.placesDisponibles = placesDisponibles;
        this.prix = prix;
        this.voiture = voiture;
        this.timestamp = timestamp;
    }

    // Constructor with only mandatory fields
    public Trajet(String villeDepart, String villeArrivee, double prix) {
        this.villeDepart = villeDepart;
        this.villeArrivee = villeArrivee;
        this.prix = prix;
    }

}

File: FrontEnd/Mobile/lib/screens/transport/models/trajet.dart
class Trajet {
  final String nomConducteur;
  final String villeDepart;
  final String villeArrivee;
  final String date;
  final String heure;
  final int placesDisponibles;
  final double prix;
  final String voiture;
  final String? photoUrl;

  Trajet({
    required this.nomConducteur,
    required this.villeDepart,
    required this.villeArrivee,
    required this.date,
    required this.heure,
    required this.placesDisponibles,
    required this.prix,
    required this.voiture,
    this.photoUrl,
  });

  factory Trajet.fromJson(Map<String, dynamic> json) {
    return Trajet(
      nomConducteur: json['nomConducteur'],
      villeDepart: json['villeDepart'],
      villeArrivee: json['villeArrivee'],
      date: json['date'],
      heure: json['heure'],
      placesDisponibles: json['placesDisponibles'],
      prix: json['prix'].toDouble(),
      voiture: json['voiture'],
      photoUrl: json['photoUrl'],
    );
  }
}